### PR TITLE
Replace messageType with type

### PR DESCRIPTION
**Pull request by marcin-cebo**
- messageType represents something like event type e.g. message, signal, file, object, messageAction; 
- type represents user defined value.

### CODE REVIEW

File: src/main/java/com/pubnub/api/enums/MessageType.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+import java.util.HashMap;
+import java.util.Map;
+
+import static com.pubnub.api.builder.PubNubErrorBuilder.PNERROBJ_UNKNOWN_MESSAGE_TYPE;
+
+public enum MessageType {
+
+    MESSAGE01(null),
+    MESSAGE02(0),
+    SIGNAL(1),         //this value is not expected in history call
```
Comment (kleewho) [inline src/main/java/com/pubnub/api/enums/MessageType.java:14]: We have a type representing what can and what cannot be expected in history call already so this comments are not needed

File: src/test/java/com/pubnub/api/models/consumer/history/PNFetchMessageItemTest.java [hunk 2]
```diff
@@ -19,33 +17,70 @@ void when_includeMessageType_is_false_then_getMessageType_should_return_null() {
...
         String spaceId = null;
         Integer pnMessageType = null;
-        String userMessageType = null;
+        String type = "myType";
 
-        objectUnderTest = getPNFetchMessageItem(includeMessageType, spaceId, pnMessageType, userMessageType);
+        objectUnderTest = getPNFetchMessageItem(includeMessageType, spaceId, pnMessageType, type);
 
         assertNull(objectUnderTest.getMessageType());
+        assertEquals(type, objectUnderTest.getType());
```
Comment (kleewho) [inline src/test/java/com/pubnub/api/models/consumer/history/PNFetchMessageItemTest.java:25]: Why test type if test name is about messageType?

File: src/test/java/com/pubnub/api/models/consumer/history/PNFetchMessageItemTest.java [hunk 3]
```diff
@@ -19,33 +17,70 @@ void when_includeMessageType_is_false_then_getMessageType_should_return_null() {
...
+        Integer eValueFromServer = null;
+        String type = null;
+
+        objectUnderTest = getPNFetchMessageItem(includeMessageType, spaceId, eValueFromServer, type);
+
+        assertNull(objectUnderTest.getType());
+    }
+
+    @Test
+    void when_includeMessageType_is_true_and_e_from_server_is_null_then_getMessageType_should_return_MESSAGE() {
```
Comment (kleewho) [inline src/test/java/com/pubnub/api/models/consumer/history/PNFetchMessageItemTest.java:41]: There's no `e` in fetchMessages. It applies to more test names

**Review by kleewho — APPROVED** [review]
I would suggest minor changes but it's fine in general

File: src/test/java/com/pubnub/api/models/consumer/history/PNFetchMessageItemTest.java (hunk 2 above)
Comment (marcin-cebo) [reply src/test/java/com/pubnub/api/models/consumer/history/PNFetchMessageItemTest.java:25]: Ok, I can create separate test to cover this case.

File: src/main/java/com/pubnub/api/models/consumer/history/HistoryMessageType.java [hunk 4]
```diff
@@ -0,0 +1,30 @@
+package com.pubnub.api.models.consumer.history;
+
+import com.pubnub.api.PubNubRuntimeException;
+
+import static com.pubnub.api.builder.PubNubErrorBuilder.PNERROBJ_UNKNOWN_HISTORY_MESSAGE_TYPE;
+
+public enum HistoryMessageType {
+    MESSAGE(0), // if includeMessageType is true then null from server also indicates MESSAGE
```
[the remark this answers is not in the document]
Comment (marcin-cebo) [reply src/main/java/com/pubnub/api/models/consumer/history/HistoryMessageType.java:8]: ok, I will modify it and move it to "of" method.
I would like to have comment i some shape to indicate that something is strange here (null indicates MESSAGE)


    public static HistoryMessageType of(Integer eValueFromServer) {
        // MESSAGE is represented by two eValueFromServer either "null" or "0".   <---------here comment
        if (eValueFromServer == null || eValueFromServer == MESSAGE.eValueFromServer) {
            return MESSAGE;
        } else if (eValueFromServer == FILE.eValueFromServer) {
            return FILE;
        } else {
            throw PubNubRuntimeException.builder().pubnubError(PNERROBJ_UNKNOWN_HISTORY_MESSAGE_TYPE).build();
        }
    }

File: src/main/java/com/pubnub/api/enums/MessageType.java (hunk 1 above)
Comment (marcin-cebo) [reply src/main/java/com/pubnub/api/enums/MessageType.java:14]: ok, makes sense.

File: src/main/java/com/pubnub/api/models/consumer/history/PNFetchMessageItem.java [hunk 5]
```diff
@@ -29,12 +27,12 @@ public class PNFetchMessageItem {
     private final boolean includeMessageType;
 
     @Getter(AccessLevel.NONE)
+    // "e" value null or 0 = message, 4 = file
```
[the remark this answers is not in the document]
Comment (marcin-cebo) [reply src/main/java/com/pubnub/api/models/consumer/history/PNFetchMessageItem.java:30]: ok, makes sense. I will remove it.

**Pull request merged**